### PR TITLE
27 create the hero section

### LINKED ISSUE #27: Create the hero section
**ACCEPTANCE CRITERIA**

We want to create a hero section that will contain maybe a background image, a title and a CTA that lead directly to openning the modal of the form. I am not sure of the figma design since i don't have access to it but that are the items that would be required.

We want to also make sure to like the `how` link of the menu to this section of the page (use #id for that in the html)

**TASKS**

- [x] Create the `hero` section linked the the `how` id.
- [x] Style the content of it
- [x] Make sure to add an `onClick` event to the `button` CTA that would lead to open the modal of the form.
- [x] The `onClick` event should trigger an handleFunction called `handleModal` that for now should modify the boolean value a hook state to its contrary (ref to what we did in the header)
```js
const [modal, setModal] = useState(false)
```
Make sure to show a `hello world` content when the `modal` state is `true`

### PR DESCRIPTION
**Pull request by Isynopoulos1**
-I implemented the modal logic when the user clicks in the link or confirm button
-I styled the hero section
-I added styling to the modal component
-I added responsive rules

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,20 +1,53 @@
...
       <GlobalStyles />
-      <Header />
-      <div style={{ margin: "35rem 0" }}>CONTENT TO DO</div>
+      <Header
+        onConfirmClick={handleConfirmed}
+        isModalOpen={isModalOpen}
+        onModalOpen={handleModalOpen}
+        onModalClose={handleModalClose}
+      />
+      <section id="when" style={{ margin: "35rem 0" }}>
```
Comment (erwanriou) [inline src/App.jsx]: - [x] remove all the section `html` from here. And make sure that each styled component you import **IS** a section
- [x] make sure to don't forgot the ID
- [x] Remove the hardcoded styles

File: src/components/atoms/when/When.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const handleClick = () => {
+    if (typeof onConfirmClick === "function") {
+      onConfirmClick();
+    }
+    if (typeof onModalOpen === "function") {
+      onModalOpen();
+    }
+  };
+  return (
+    <>
```
Comment (erwanriou) [inline src/components/atoms/when/When.jsx]: Here is probably where you should have your `section` styled or not in html

**Review by erwanriou — CHANGES_REQUESTED** [review]
A few things to update

- [x] Remove the unecessary state on the `App.js`
- [x] Make sure the there only one `setter` function passed through out the 2 children components
- [x] Make sure the modal is not been loaded from the `header` but from the `App` instead
- [x] Fix the section styles
- [x] Remove unused props and functions
- [x] Make sure to add the CTA form button to the mobile menu
- [x] Fix the onClick event of the link on the mobile menu

**Comment by Isynopoulos1** [comment]
I fixed the bugs & is ready to review

File: src/App.jsx [hunk 3]
```diff
@@ -1,21 +1,36 @@
...
+  const handleModal = () => {
+    return setModalOpen(!isModalOpen);
+  };
+
   return (
     <>
       <GlobalStyles />
-      <Header />
-      <div style={{ margin: "35rem 0" }}>CONTENT TO DO</div>
+      <Header isModalOpen={isModalOpen} toggleModal={handleModal} />
```
Comment (erwanriou) [inline src/App.jsx]: are you using the `isModalOpen` in `Header`?

File: src/App.jsx [hunk 4]
```diff
@@ -1,21 +1,36 @@
...
+      <When onModalOpen={handleModal} />
+      <Where />
+      <How />
+
       <Footer />
+      {isModalOpen && <Modal onClose={handleModal} />}
```
Comment (erwanriou) [inline src/App.jsx]: for the sake of readability i would but that line just on top of the `Header` line 26

File: src/components/atoms/confirmBtn/ConfirmBtn.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { BtnContainer } from "./ConfirmBtn.styles";
+
+const ConfirmBtn = ({ onClick, onModalOpen }) => {
+  const handleClick = () => {
```
Comment (erwanriou) [inline src/components/atoms/confirmBtn/ConfirmBtn.jsx]: you don't need to use this function and you definitly don't need to use 2 functions here. Please refactor

File: src/components/atoms/when/When.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+//IMPORT STYLES
+
+import { HeroContainer, InfoContainer, MainContainer } from "./When.styles";
+//IMPORT COMPONENTS
+import Logo from "@atoms/logo/mobile/MobileLogo";
+
+export const When = ({ onConfirmClick, onModalOpen }) => {
+  const handleClick = () => {
```
Comment (erwanriou) [inline src/components/atoms/when/When.jsx]: You are not using this function so why is it here.

File: src/components/atoms/when/When.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+//IMPORT STYLES
+
+import { HeroContainer, InfoContainer, MainContainer } from "./When.styles";
+//IMPORT COMPONENTS
+import Logo from "@atoms/logo/mobile/MobileLogo";
+
+export const When = ({ onConfirmClick, onModalOpen }) => {
```
Comment (erwanriou) [inline src/components/atoms/when/When.jsx]: where are you using `onConfirmClick`

**Review by erwanriou — APPROVED** [review]
It work but i added some few thing to remove as not used.

File: src/App.jsx (hunk 3 above)
Comment (Isynopoulos1) [reply src/App.jsx]: done

File: src/App.jsx (hunk 4 above)
Comment (Isynopoulos1) [reply src/App.jsx]: done

File: src/components/atoms/confirmBtn/ConfirmBtn.jsx (hunk 5 above)
Comment (Isynopoulos1) [reply src/components/atoms/confirmBtn/ConfirmBtn.jsx]: done

File: src/components/atoms/when/When.jsx (hunk 6 above)
Comment (Isynopoulos1) [reply src/components/atoms/when/When.jsx]: done

File: src/components/atoms/when/When.jsx (hunk 7 above)
Comment (Isynopoulos1) [reply src/components/atoms/when/When.jsx]: done

**Comment by Isynopoulos1** [comment]
I made all the changes, let me know if I can merge it please

**Comment by erwanriou** [comment]
All good for merging!

**Pull request merged**